src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs

```diff
@@ -544,10 +544,10 @@ private void WriteArray(WriteObjectInfo objectInfo, NameInfo memberNameInfo, Wri
 
                 _serWriter.WriteJaggedArray(memberNameInfo, arrayNameInfo, arrayElemObjectInfo, arrayElemTypeNameInfo, lengthA[0], lowerBoundA[0]);
 
-                var objectA = (object[])array;
+                var objectA = (Array)array;
                 for (int i = lowerBoundA[0]; i < upperBoundA[0] + 1; i++)
                 {
-                    WriteArrayMember(objectInfo, arrayElemTypeNameInfo, objectA[i]);
+                    WriteArrayMember(objectInfo, arrayElemTypeNameInfo, objectA.GetValue(i));
```

Comment:
Does this have any measurable impact on the performance of serializing/deserializing arrays?  Or is this only hit for the (very uncommon) case of non-zero-lower-bound arrays?